libs/deposits/src/lib/deposits-table.tsx

```diff
@@ -0,0 +1,75 @@
+import type {
+  ICellRendererParams,
+  ValueFormatterParams,
+} from 'ag-grid-community';
+import { AgGridColumn } from 'ag-grid-react';
+import {
+  t,
+  addDecimalsFormatNumber,
+  getDateTimeFormat,
+  truncateByChars,
+} from '@vegaprotocol/react-helpers';
+import { AgGridDynamic as AgGrid, Link } from '@vegaprotocol/ui-toolkit';
+import type { DepositFields } from './__generated__/DepositFields';
+import { useEnvironment } from '@vegaprotocol/environment';
+import type { DepositStatus } from '@vegaprotocol/types';
+import { DepositStatusMapping } from '@vegaprotocol/types';
+
+export interface DepositsTableProps {
+  deposits: DepositFields[];
+}
+
+export const DepositsTable = ({ deposits }: DepositsTableProps) => {
```

Comment:
Ag-grid tables can be typed little bit more. This will allow to catch changes in data used by table.

```
import type {
  ICellRendererParams,
  ValueFormatterParams,
  IGetRowsParams,  
  IDatasource
} from 'ag-grid-community';
import type { AgGridReactProps } from 'ag-grid-react';
import type { Deposits_party_depositsConnection_edges_node } from './__generated__/Deposits';


...


export interface GetRowsParams extends Omit<IGetRowsParams, 'successCallback'> {
  successCallback(rowsThisBlock: Deposits_party_depositsConnection_edges_node[], lastRow?: number): void;
}

export interface Datasource extends IDatasource {
  getRows(params: GetRowsParams): void;
}
interface Props extends AgGridReactProps {
  rowData?: Deposits_party_depositsConnection_edges_node[] | null;
  datasource?: Datasource;
}

type PositionsTableValueFormatterParams = Omit<
  ValueFormatterParams,
  'data' | 'value'
> & {
  data: Deposits_party_depositsConnection_edges;
};


export const DepositsTable = (props: Props) => {
  const { ETHERSCAN_URL } = useEnvironment();
  return (
    <AgGrid
      overlayNoRowsTemplate={t('No deposits')}
      defaultColDef={{ flex: 1, resizable: true }}
      style={{ width: '100%', height: '100%' }}
      suppressCellFocus={true}
      {...props}
    >
    
    
    ....
    
    <AgGridColumn
        headerName="Amount"
        field="amount"
        valueFormatter={({ value, data }: PositionsTableValueFormatterParams & { value: Deposits_party_depositsConnection_edges_node_asset["symbol"] }) => {
          return addDecimalsFormatNumber(value, data.asset.decimals);
        }}
      />
    



```

Reply:
@bglownia I've added some helper types for this in libs/ui-toolkit/src/components/ag-grid/index.tsx, take a look. Perhaps we might want to add them to use in other tables.